### ISSUE TITLE
Compatible Component Libraries

### ISSUE THREAD

**Issue by strandedcity** [solidjs/solid#646]
Hi There,

I was wondering if there's a rundown somewhere of the state of component libraries that are compatible with solid? I've done some searching, and come up short. Here's the situation:

1. I'm totally sold on Solid's approach. I'm considering it as a replacement for a large legacy react app whose main drivers are performance, and control over the render cycle. Secondarily, however, there is quite a lot of stateful ui (buttons, pickers, etc) and the solution will need to scale with a team. Solid seems excellent, but I'm stumped trying to figure out the best compatible solution for, say, a date picker. (I mean this as an example -- a link to an atomic date picker that will work is not what I'm after. Really I'm looking for a themeable drawer of pre-rolled components that we can use out of the box, and ideally customize when necessary)
2. The examples in the solid project seem to borrow css from somewhere, and use it narrowly within the scope of the demo. Eg, using `todomvc-app-css` is a solution that does not scale beyond the TODO example.
3. All the big UI libs (material, semantic, bootstrap, etc) seem to fall into one of two categories:

- A big project full of react bindings for things like toasts and datepickers (which presumably can't just be imported into solid). Sometimes (in semantic, for example) there are bindings for other frameworks also ... angular, meteor, ember, but never solid.
- A big project full of jQuery-compatible css/js components that could be imported, and then be available in solid. In this case, JSX project files would get filled with the html markup particular to the library, so a `<Button>` is now some 50-line mess of markup that's hard-tied to the UI library, and repeated all over the project code base. For a button of course we'd end up rolling a custom Solid `<Button>` component, but for a datepicker this would be painful. Also, jQuery seems like the past not the future.

What's the recommendation? How is it possible to build a scalable team around SolidJS, even if SolidJS is the perfect solution for the core problem? I realize this is an "ecosystem" problem, but I'd take even a strategy for contributing to such a library ... a team comes along, needs a datepicker, and builds that component for Semantic/Solid, for example ... but we'd probably need a place to start.

All advice welcome!

**Comment by mosheduminer** [comment]
Hey @strandedcity :wave:,

There's development around component libraries, which can be seen in the `#libraries` channel on the solid discord.

In terms of "the big UI libs", we're working on a port of carbon-components-react (which also has datepickers 🙂) to Solid. So all functionality is written with Solid, and css depends on the sass style sheets from the vanilla carbon-components package (which is how the react and svelte libs do it as well). I'm about 1/3 of the way through (we're talking about some 90 components). We also have an offer to turn it into an official community maintained at Carbon.

[Unfortunately, all this work takes time, and the repo is currently private do its vastly underfinished state. Which is why I'll take a moment to plug that if you just so happen to have time to spare, `more help` => `faster release` 😉.]

**Comment by atk** [comment]
I'm preparing an original component library for solid.js: https://atk.github.io/solid-blocks/

I'm still trying to nail isomorphic components (client-side and hydrated). Once I got that running, I'll release a first version.

**Comment by strandedcity** [comment]
Thank you both for the responses, super helpful ! Both options look ... solid. 

I would love to contribute to one of the efforts, but if I'm honest with myself the setup and push to learn and use solid internally is going to be pushing the limits of the time and energy available to me and my team right now. It's great to see the community coming together around it! 

A possible temporary solution out there... I did take another look at Bootstrap 5 while browsing around, and learned that they did the work to turn it into a vanilla JS library. Obviously this doesn't get us all the way to Soild components, but it does make the use of Bootstrap in a temporary way a little more palatable.

Discord had some weird errors when I tried to sign up (something about getting rate limited? I promise it was just me...) but I'll join the discussion when I can! To keep the issues section clean, I'll go ahead and close this one. Thanks again.
